numpy/core/tests/test_numeric.py

```diff
@@ -2018,7 +2018,8 @@ def test_NaT_propagation(self, arr, amin, amax):
         # NOTE: the expected function spec doesn't
         # propagate NaT, but clip() now does
         expected = np.minimum(np.maximum(arr, amin), amax)
-        actual = np.clip(arr, amin, amax)
+        with assert_warns(DeprecationWarning):
+            actual = np.clip(arr, amin, amax)
```

Comment:
How about `@pytest.mark.filterwarnings("ignore::DeprecationWarning")`? Using assert_warns seems out of place for something that will be changed in the future, tests for warnings belong in `test_deprecations`.

Reply:
Done. Another option is to delete this test: I think it started out as an edge case reported by hypothesis, and has now both an xfail and a warning filter.